File: CS/Q205815/MyGridControl.cs
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Printing;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrinting.Native;
using System.Drawing;

namespace DXSample {
    public class MyGridControl : GridControl {
        public MyGridControl() : base() { }
        protected override void RegisterAvailableViewsCore(InfoCollection collection) {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new MyGridViewInfoRegistrator());
        }
    }
    public class MyGridView : GridView {
        public MyGridView() : base() { }
        public MyGridView(GridControl gridControl) : base(gridControl) { }
        internal const string MyGridViewName = "MyGridView";
        protected override string ViewName { get { return MyGridViewName; } }
        protected override BaseViewPrintInfo CreatePrintInfoInstance(PrintInfoArgs args) {
            return new MyGridViewPrintInfo(args);
        }
    }
    public class MyGridViewInfoRegistrator : GridInfoRegistrator {
        public MyGridViewInfoRegistrator() : base() { }
        public override string ViewName { get { return MyGridView.MyGridViewName; } }
        public override BaseView CreateView(GridControl grid) {
            return new MyGridView(grid);
        }
    }
    public class MyGridViewPrintInfo : GridViewPrintInfo {
        public MyGridViewPrintInfo(PrintInfoArgs args) : base(args) { }
        protected override void PrintGroupRow(int rowHandle, int level) {
            Rectangle r = Rectangle.Empty;
            r.X = Indent + level * ViewViewInfo.LevelIndent;
            r.Width = this.fMaxRowWidth - r.Left;
            r.Y = Y;
            r.Height = CurrentRowHeight;
            SetDefaultBrickStyle(Graph, Bricks["GroupRow"]);
            Brick brick = (Brick)DrawTextBrick(Graph, View.GetGroupRowDisplayText(rowHandle), r, true);
            if (Y != 0)
                ((PSLinkDocument)((PrintingSystemBase)PS).Document).ShowFromNewPage(brick);
            Y += r.Height;
        }
    }
}

File: VB/Q205815/MyGridControl.vb
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Printing
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrinting.Native
Imports System.Drawing

Namespace DXSample

    Public Class MyGridControl
        Inherits GridControl

        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As InfoCollection)
            MyBase.RegisterAvailableViewsCore(collection)
            collection.Add(New MyGridViewInfoRegistrator())
        End Sub
    End Class

    Public Class MyGridView
        Inherits GridView

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal gridControl As GridControl)
            MyBase.New(gridControl)
        End Sub

        Friend Const MyGridViewName As String = "MyGridView"

        Protected Overrides ReadOnly Property ViewName As String
            Get
                Return MyGridViewName
            End Get
        End Property

        Protected Overrides Function CreatePrintInfoInstance(ByVal args As PrintInfoArgs) As BaseViewPrintInfo
            Return New MyGridViewPrintInfo(args)
        End Function
    End Class

    Public Class MyGridViewInfoRegistrator
        Inherits GridInfoRegistrator

        Public Sub New()
            MyBase.New()
        End Sub

        Public Overrides ReadOnly Property ViewName As String
            Get
                Return MyGridView.MyGridViewName
            End Get
        End Property

        Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
            Return New MyGridView(grid)
        End Function
    End Class

    Public Class MyGridViewPrintInfo
        Inherits GridViewPrintInfo

        Public Sub New(ByVal args As PrintInfoArgs)
            MyBase.New(args)
        End Sub

        Protected Overrides Sub PrintGroupRow(ByVal rowHandle As Integer, ByVal level As Integer)
            Dim r As Rectangle = Rectangle.Empty
            r.X = Indent + level * ViewViewInfo.LevelIndent
            r.Width = Me.fMaxRowWidth - r.Left
            r.Y = Y
            r.Height = CurrentRowHeight
            SetDefaultBrickStyle(Graph, Bricks("GroupRow"))
            Dim brick As Brick = CType(DrawTextBrick(Graph, View.GetGroupRowDisplayText(rowHandle), r, True), Brick)
            If Y <> 0 Then CType(CType(PrintingSystemBase, PrintingSystemBase).Document, PSLinkDocument).ShowFromNewPage(brick)
            Y += r.Height
        End Sub
    End Class
End Namespace
